packages/toolkit/src/entities/unsorted_state_adapter.ts

```diff
@@ -158,7 +161,7 @@ export function createUnsortedStateAdapter<T>(
         0
 
       if (didMutateIds) {
-        state.ids = Object.keys(state.entities)
+        state.ids = Object.keys(state.entities) as Id[]
```

Comment:
hmm - this isn't actually a safe assertion.

Object.keys *always* returns string[], even if numbers were used as keys:
![image](https://github.com/reduxjs/redux-toolkit/assets/18308300/0fbef37e-07a0-45eb-bb65-94da58b02a59)

not entirely sure how to handle this better though

Reply:
Fixed for now. I think the 'proper' way is to enforce ids to be strings only - as that is what is actually going on here. Not sure what @markerikson thinks about that, but the current way clearly has some issues.